lib/logstash/inputs/snmp.rb

```diff
@@ -132,7 +151,7 @@ def register
       # TODO: move these validations in a custom validator so it happens before the register method is called.
       host_details = host_name.match(HOST_REGEX)
       raise(LogStash::ConfigurationError, "invalid format for host option '#{host_name}'") unless host_details
-      raise(LogStash::ConfigurationError, "only udp & tcp protocols are supported for host option '#{host_name}'") unless host_details[:host_protocol].to_s =~ /^(?:udp|tcp)$/i
+      raise(LogStash::ConfigurationError, "unsupported protocol for host option '#{host_name}'") unless host_details[:host_protocol].to_s =~ /^(?:udp|tcp|tls)$/i
```

Comment:
is adding support for tls this easy? Looking at the client, I'm seeing that we send [`BaseClient#create_transport(protocol)`](https://github.com/logstash-plugins/logstash-input-snmp/blob/v1.2.7/lib/logstash/inputs/snmp/base_client.rb#L176), which will raise if it encounters anything other than `tcp` or `udp`. What am I missing?